src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
```

Comment:
Please apply the comments above. Use `reduce` instead of let and `forEach` (as it is mentioned [here](https://mate-academy.github.io/style-guides/javascript.html#21-useconstfor-all-of-your-references-avoid-usingvar) we predominantly use  `const`). Use full variables names: p => person.